### PR TITLE
First boot service implementation [REVPI-3063]

### PR DESCRIPTION
**Pull request by RamiGspo**
In order to implement the "firstboot.service" some modifications in revpi-tools are needed.

### CODE REVIEW

**Review by iluminat23 — COMMENTED** [review]
In commit https://github.com/RevolutionPi/revpi-tools/pull/28/commits/1f94cbfb4268976a30f29bb02ade1d86a550ece8 the commit message contains this `\fBrevpi-factory-reset\fP`. I guess this should be some form of formatting. Please keep in mind, commit messages are plain text no formatting is possible.

File: firstboot/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+install(FILES firstboot.service
+  DESTINATION /etc/systemd/system/
```
Comment (naruxde) [inline firstboot/CMakeLists.txt:2]:
```suggestion
  DESTINATION ${CMAKE_INSTALL_ SYSCONFDIR}/systemd/system/
```
Do we want /etc or /lib/systemd?

The main problem will end up in the Debian package. That should use dh_installsystemd and needs the files in the Debian directory. Has anyone tested whether it makes any sense to integrate the unit files into cMake?

**Review by naruxde — CHANGES_REQUESTED** [review]

File: firstboot/CMakeLists.txt (hunk 1 above)
Comment (naruxde) [reply firstboot/CMakeLists.txt:2]: Other .service files of this project are in "/lib/systemd/system". We should check Debian branch, how the systemd services are handled. If there is no dh magic, we have to write our own postins scripts :(

Comment (nbuchwitz) [reply firstboot/CMakeLists.txt:2]: https://unix.stackexchange.com/a/208352

Comment (iluminat23) [reply firstboot/CMakeLists.txt:2]: /etc/systemd is for the user and where the symlinks are created to enable/disable units. Never install a systemd file into /etc/systemd.

**Review by iluminat23 — CHANGES_REQUESTED** [review]

**Comment by iluminat23** [comment]
It is unclear to me how to proceed when we want to add more stuff to the first boot service. In the future we also want to generate ssh-keys. We want to create snakeoil certificates, ...
Do we need to modify the `firstboot.sh`? which seems rather messy if we put all the code in this script.

**Review by iluminat23 — APPROVED** [review]

**Comment by RamiGspo** [comment]
@iluminat23 

Because of [this PR](https://github.com/RevolutionPi/revpi-tools/pull/30) I had to rebase to avoid the conflicts

File: firstboot/resize-fs.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# resize root partition
+dev=mmcblk0
+total_size="$(/bin/cat /sys/block/$dev/size)"
+last_part="$(cd /dev || exit 1 ; /bin/ls ${dev}p* | /usr/bin/tail -1)"
+last_part_start="$(/bin/cat "/sys/block/$dev/$last_part/start")"
+last_part_size="$(/bin/cat "/sys/block/$dev/$last_part/size")"
+last_part_max="$(("$total_size" - "$last_part_start"))"
+if [ "$last_part_size" -lt "$last_part_max" ] ; then
+	parted "/dev/$dev" resizepart 2 $((total_size-1))s
```
Comment (naruxde) [inline firstboot/resize-fs.sh:11]:
```suggestion
	/usr/sbin/parted "/dev/$dev" /usr/bin/resizepart 2 $((total_size-1))s
```
Why ist parted not absolut? All other commands are.

**Review by naruxde — CHANGES_REQUESTED** [review]
Keep in mind to check dependencies in debian branches for: parted, partprobe, resize2fs, resizepart

File: firstboot/resize-fs.sh (hunk 2 above)
Comment (RamiGspo) [reply firstboot/resize-fs.sh:11]: Is it really necessary?
The service works the way it is now.. Of course I can modify it, but do we really need it?

**Review by naruxde — APPROVED** [review]

File: firstboot/resize-fs.sh (hunk 2 above)
Comment (RamiGspo) [reply firstboot/resize-fs.sh]: @naruxde @nbuchwitz 
Thank you

**Pull request merged**